tests-trigger

```diff
@@ -10,20 +11,26 @@ from task import github
 
 
 def trigger_pull(api, opts):
-    pull = api.get("pulls/" + opts.pull)
-
-    if not pull:
-        sys.stderr.write("{0} is not a pull request.\n".format(opts.pull))
-        return 1
-
-    # triggering is manual, so don't prevent triggering a user that does not have push access
-    # nor isn't in the 'Contributors' group, but issue a warning in case of an oversight
-    login = pull["head"]["user"]["login"]
-    if not opts.allow and not api.is_user_allowed(login):
-        sys.stderr.write("Pull request author '{0}' isn't allowed. Override with --allow.\n".format(login))
-        return 1
+    if opts.pull != '-':
+        pull = api.get("pulls/" + opts.pull)
+
+        if not pull:
+            sys.stderr.write("{0} is not a pull request.\n".format(opts.pull))
+            return 1
+
+        # triggering is manual, so don't prevent triggering a user that does not have push access
+        # nor isn't in the 'Contributors' group, but issue a warning in case of an oversight
+        login = pull["head"]["user"]["login"]
+        if not opts.allow and not api.is_user_allowed(login):
+            sys.stderr.write("Pull request author '{0}' isn't allowed. Override with --allow.\n".format(login))
+            return 1
+
+        revision = pull['head']['sha']
+        target = f'pull request {opts.pull}'
+    else:
+        revision = subprocess.check_output(['git', 'rev-parse', '@{u}'], encoding='utf-8').strip()
```

Comment:
I suggest to use the long name `{upstream}` to make this a bit easier to read.

Reply:
done, thanks